external/torch-mlir/lib/Dialect/Torch/IR/TorchTypes.cpp

```diff
@@ -220,10 +220,15 @@ NonValueTensorType::getWithLeastStaticInformation(MLIRContext *context) {
 }
 
 NonValueTensorType NonValueTensorType::getFromShaped(ShapedType type) {
-  return NonValueTensorType::get(type.getContext(),
-                                 type.hasRank() ? type.getShape()
-                                                : Optional<ArrayRef<int64_t>>(),
-                                 type.getElementType());
+  Type dtype = type.getElementType();
```

Comment:
This is not safe in general, since there is no way to know whether the type we want is signed or unsigned. For example, int8 can be signed or unsigned
https://pytorch.org/docs/stable/tensors.html

Can you just remove this method? We might need to update a few places, but generally it's really sketchy to do this and it seems like a good direction to remove the method. Same for ValueTensorType::getFromShaped